### ISSUE TITLE
[ENH] Compilation does not support importing cython module from another module

### ISSUE THREAD

**Issue by matusvalo** [cython/cython#6129]
### Is your feature request related to a problem? Please describe.

Consider following situation:
```python
######## py_module.py #######
import cython
####### cy_module.py ########
from py_module import cython

def main():
    a: cython.int = 5
    print(a + 1)
```

When compiling Cython ignores `cython.int` annotation because it does not recognise `cython` module even though from Python semantics the correct cython module is imported.

### Describe the solution you'd like.

Importing cython module recognise cython module during compilation even not imported directly. The behaviour will match python semantics.

### Describe alternatives you've considered.

I suppose implementing this issue 100% correctly is complex - cython needs to execute real import and check whether cython module is cython's one. Maybe the initial sufficient solution is just to consider cython module imported when `cython` name is present in import:
* `from dummy import cython` 
* `from dummy.subdummy import cython`
* `from dummy.cython import int`

In all these cases, compiler will consider imported cython module even though it is never checked what is inside module.

Of coarse this logic will apply only for compilation and should be explicitly documented.

### Additional context

This feature seems to be desired from users of cython - see:
* https://github.com/falconry/falcon/issues/2189#issuecomment-1839723159
* https://github.com/sqlalchemy/sqlalchemy/blob/d3222a31b8df97a454b37a32881dd484a06e5742/lib/sqlalchemy/engine/_processors_cy.py#L16-L24

**Comment by da-woods** [comment]
I think https://github.com/sqlalchemy/sqlalchemy/blob/d3222a31b8df97a454b37a32881dd484a06e5742/lib/sqlalchemy/engine/_processors_cy.py#L16-L24 is more related to https://github.com/cython/cython/issues/1981

What they're really looking for is the ability to work uncompiled and in environments without Cython (where `import cython` will fail completely)

**Comment by da-woods** [comment]
Looking at the "Falconry" issue I think they're trying to do the same thing and essentially provide their own copy of "Shadow.py". I can see why they're doing it, but I still think the right solution is to distribute "Shadow.py" as a lightweight standalone package that people are happy to have as a runtime requirement rather than to try to follow imports through Python modules.

**Comment by scoder** [comment]
I agree that we should rather do https://github.com/cython/cython/issues/1981 than tracking arbitrary (runtime) imports across modules, including modules that are not even compiled, that might not be available at build time, or that happen to have the word `cython` in them.

**Comment by matusvalo** [comment]
Now when I am looking at it again, I agree with @da-woods and @scoder. I think closing #1981 will eventually close this issue.
